.github/workflows/build-cuda-wheels.yml

```diff
@@ -0,0 +1,98 @@
+# inspired by https://github.com/AutoGPTQ/AutoGPTQ/blob/main/.github/workflows/build_wheels_cuda.yml
+
+name: Build Python wheels with CUDA
+
+on:
+  push:
+    branches: [main]
+    tags: ["*"]
+  pull_request:
+    # Check all PR
```

Comment:
Do we really want to build all CUDA wheels on all PR? I'm worried this will increase CI time a lot